src/middleware/persist.ts

```diff
@@ -19,8 +19,14 @@ export interface PersistStorage<S> {
   removeItem: (name: string) => void | Promise<void>
 }
 
+type JsonStorageOptions = {
+  reviver: (key: string, value: unknown) => unknown
+  replacer: (key: string, value: unknown) => unknown
```

Comment:
I think we can make these optional:
```suggestion
  reviver?: (key: string, value: unknown) => unknown
  replacer?: (key: string, value: unknown) => unknown
```